plugin/trino-opa/src/main/java/io/trino/plugin/opa/schema/TrinoIdentity.java

```diff
@@ -16,24 +16,29 @@
 import com.google.common.collect.ImmutableSet;
 import io.trino.spi.security.Identity;
 
+import java.security.Principal;
+import java.util.Optional;
 import java.util.Set;
 
 import static java.util.Objects.requireNonNull;
 
 public record TrinoIdentity(
         String user,
-        Set<String> groups)
+        Set<String> groups,
+        Optional<Principal> principal)
```

Comment:
I checked how these are sent and it seems the `OpaHttpClient` serializes everything to JSON. Are we sure that all principals can be serialized properly?

I don't have a really good answer myself for how to test this to be honest.

Reply:
Yeah, we only have the `java.security.Principal` interface here. In Rust I would simply add a trait bound on `Serialize` and the compiler would ensure all passed types can be serialized :P
This is the main reason this is opt-in, so only users that really need access to the principal in OPA need to make sure it's serializable.

It would be great to find out what possible concrete types the Trino source code passes to authorizers.